app/ui/legacy/src/main/java/com/fsck/k9/ui/messagelist/MessageListSwipeCallback.kt

```diff
@@ -288,13 +291,29 @@ class MessageListSwipeCallback(
         return (super.getAnimationDuration(recyclerView, animationType, animateDx, animateDy) * percentage).toLong()
     }
 
-    private fun setupSwipeAction(swipeAction: SwipeAction, resourceProvider: SwipeResourceProvider) = SwipeActionConfig(
-        colorRoles = resourceProvider.getActionColorRoles(swipeAction),
-        icon = resourceProvider.getActionIcon(swipeAction),
-        iconToggled = resourceProvider.getActionIconToggled(swipeAction),
-        actionName = resourceProvider.getActionName(swipeAction),
-        actionNameToggled = resourceProvider.getActionNameToggled(swipeAction),
-    )
+    private fun isSwipeActionSupported(item: MessageListItem, swipeAction: SwipeAction): Boolean {
+        return when (swipeAction) {
+            SwipeAction.None -> false
```

Comment:
Maybe at least a comment if there is no unit test to avoid this gets refactored in the future again?

Reply:
Introducing this function is not necessary to fix the crash. I'll remove it to keep this change focused on fixing the immediate cause of the crash.

We'll need to refactor `MessageListSwipeCallback`, `SwipeResourceProvider`, and `SwipeActionSupportProvider` anyway to avoid a problem like this crash in the future.

Reply:
Yes. I thought so two. The function refactoring could be a separate commit at best.